src/CTA.WebForms/Services/TaskManagerService.cs

```diff
@@ -93,27 +106,24 @@ private void UpdateStallState()
             // then we do nothing
         }
 
-        private void TryResolveStall()
+        private async void TryResolveStall()
         {
-            // NOTE: Commented out for use later, currently
-            // configured to immediately cancel oldest task
-            // on stall
+            var currentStallOccurrence = _stallOccurrenceNumber;
+            await Task.Delay((int)_stallTimeoutMs);
```

Comment:
this will hit application performance? why we need to delay the task?

Reply:
Also, if the stall resolves during the wait time, it should actually have no impact on performance at all since the application continues to run normally during this delay